src/main/scala/com/lucidworks/spark/SolrRelation.scala

```diff
@@ -99,17 +99,8 @@ class SolrRelation(
   override def buildScan(fields: Array[String], filters: Array[Filter]): RDD[Row] = {
 
     if (fields != null && fields.length > 0) {
-      // If all the fields in the base schema are here, we probably don't need to explicitly add them to the query
-      if (this.baseSchema.size == fields.length) {
-        // Special case for docValues. Not needed after upgrading to Solr 5.5.0 (unless to maintain back-compat)
-        if (conf.docValues.getOrElse(false)) {
-          fields.zipWithIndex.foreach({ case (field, i) => fields(i) = field.replaceAll("`", "") })
-          query.setFields(fields: _*)
-        }
-      } else {
-        fields.zipWithIndex.foreach({ case (field, i) => fields(i) = field.replaceAll("`", "") })
-        query.setFields(fields: _*)
-      }
+      fields.zipWithIndex.foreach({ case (field, i) => fields(i) = field.replaceAll("`", "") })
```

Comment:
I thought it is unnecessary to add list of fields to every request. I will take a look at why the test is failing without this